ui/src/wallet.ts

```diff
@@ -96,28 +96,29 @@ export function build(): Wallet {
   }
 
   async function disconnect() {
-    await walletConnect.killSession();
-    // We need to reinitialize `WalletConnect` until this issue is fixed:
-    // https://github.com/WalletConnect/walletconnect-monorepo/pull/370
-    walletConnect = new WalletConnect({
-      bridge: "https://bridge.walletconnect.org",
-      qrcodeModal: qrCodeModal,
-    });
-    signer.walletConnect = walletConnect;
-    stateStore.set({ status: Status.NotConnected });
+    try {
+      stateStore.set({ status: Status.NotConnected });
+      await walletConnect.killSession();
+      // We need to reinitialize `WalletConnect` until this issue is fixed:
+      // https://github.com/WalletConnect/walletconnect-monorepo/pull/370
+      walletConnect = new WalletConnect({
+        bridge: "https://bridge.walletconnect.org",
+        qrcodeModal: qrCodeModal,
+      });
+      signer.walletConnect = walletConnect;
+    } catch {
+      // When the user disconnects wallet-side, calling `killSession` app-side trows an error
+      // because the wallet has already closed its socket. Therefore, we simply ignore it.
```

Comment:
Can we wrap this around `await walletConnect.killSession()` then. Also, is it on purpose, that in this case we don’t reinitialize `walletConnect` and signer.walletConnect`?

Reply:
Done! Not on purpose, it was an oversight on my part. Everything seemed to work well without it though, but I have addressed that as well :+1: thank you!